### PR TITLE
Improve redirection by adding top level resources

### PR DESCRIPTION
**Pull request by josepjaume**
#### :tophat: What? Why?
This updates redirections by adding top-level resources and, in the same time, constraints it to the `decidim.barcelona` domain.

#### :pushpin: Related Issues
* Depends on: https://github.com/AjuntamentdeBarcelona/decidim/pull/1012

#### :clipboard: Subtasks
*None*

### :camera: Screenshots (optional)
*None*

### CODE REVIEW

**Comment by josepjaume** [comment]
Maybe @oriolgual. Let me see what I can do.

**Comment by josepjaume** [comment]
@oriolgual done, but depends on: https://github.com/AjuntamentdeBarcelona/decidim/pull/1012

File: spec/support/database_cleaner.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    DatabaseCleaner.strategy = :transaction
+    DatabaseCleaner.clean_with(:truncation)
+  end
+
+  config.around(:each) do |example|
+    DatabaseCleaner.cleaning do
+      example.run
+    end
+  end
+
```
Comment (oriolgual) [inline spec/support/database_cleaner.rb]: Extra new line xD

File: spec/routes/redirections_spec.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
+require "rails_helper"
+require "decidim/dev/test/spec_helper"
+require "decidim/core/test/factories"
+require "decidim/proposals/test/factories"
+
+describe "routing redirections", type: :request do
```
Comment (oriolgual) [inline spec/routes/redirections_spec.rb]: Type should be routing. Although if the folder was named `routing` you wouldn't need to specify iy.

Comment (josepjaume) [inline spec/routes/redirections_spec.rb]: It turns out that `type: :routing` can't handle the `host!` option or the `redirect_to` assertion. It probably works at a lower level, but we actually need to test it full stack.

Comment (oriolgual) [inline spec/routes/redirections_spec.rb]: Can we move it to spec/requests then?

Comment (josepjaume) [inline spec/routes/redirections_spec.rb]: nnyeneyneyneye

Comment (josepjaume) [inline spec/routes/redirections_spec.rb]: Actually I don't think so, it's a routing request *hacked* so I can test what I want. Moving it to `requests` might be misleading.

**Pull request merged**